packages/autocomplete-plugin-algolia-insights/src/createSearchInsightsApi.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  ClickedFiltersParams,
+  ClickedObjectIDsAfterSearchParams,
+  ClickedObjectIDsParams,
+  ConvertedFiltersParams,
+  ConvertedObjectIDsAfterSearchParams,
+  ConvertedObjectIDsParams,
+  InsightsClient,
+  ViewedFiltersParams,
+  ViewedObjectIDsParams,
+} from './types';
+
+export function createSearchInsightsApi(searchInsights: InsightsClient) {
+  return {
+    init(appId: string, apiKey: string) {
+      searchInsights('init', { appId, apiKey });
+    },
+    setUserToken(userToken: string) {
+      searchInsights('setUserToken', userToken);
+    },
+    clickedObjectIDsAfterSearch(
```

Comment:
not sure if this makes a big difference, but this could be generated by looping over a list of method names too, wdyt?

Reply:
Wdym?

Reply:
```
const res = {}

['clickedObjectIDsAfterSearch'].forEach(method => { res[method] = (...args) => searchInsights[method] }
```

roughly this, although I just realise you lose typescript if you conditionally make the object like this, so that's probably not what you want

Reply:
Yeah, besides I'm really not a fan of "generated" APIs.

Having TS types is super useful here, especially that they're not even available in the official Search Insights lib.